src/orbit.rs

```diff
@@ -138,47 +177,89 @@ pub async fn create_orbit(
         .await
         .map_err(|e| anyhow!("Couldn't create dir: {}", e))?;
 
-    // create default and write
-    let md = OrbitMetadata {
-        id: oid.clone(),
-        controllers: policy.controllers,
-        read_delegators: vec![],
-        write_delegators: vec![],
-        revocations: vec![],
+    let (method, params) = get_oid_matrix_params(uri)?;
+
+    let md = match method {
+        "tz" => params_to_tz_orbit(oid, &params.collect::<Vec<(&str, &str)>>(), tzkt_api).await?,
+        _ => return Err(anyhow!("Unsupported method type: {}", method)),
     };
+
     fs::write(dir.join("metadata"), serde_json::to_vec_pretty(&md)?).await?;
     fs::write(dir.join("access_log"), auth).await?;
 
-    Ok(Some(load_orbit(oid, path).await.map(|o| {
+    Ok(Some(load_orbit(oid, path, key_pair).await.map(|o| {
         o.ok_or_else(|| anyhow!("Couldn't find newly created orbit"))
     })??))
 }
 
-pub async fn load_orbit(oid: Cid, path: PathBuf) -> Result<Option<SimpleOrbit>> {
+pub async fn load_orbit(
+    oid: Cid,
+    path: PathBuf,
+    key_pair: &Keypair,
+) -> Result<Option<SimpleOrbit>> {
     let dir = path.join(oid.to_string_of_base(Base::Base58Btc)?);
     if !dir.exists() {
         return Ok(None);
     }
-    load_orbit_(oid, dir).await.map(|o| Some(o))
+    load_orbit_(oid, dir, key_pair.into())
+        .await
+        .map(|o| Some(o))
+}
+
+struct KP(pub Keypair);
```

Comment:
Same questions as `PeerId`.

And `Keypair` doesn't resolve in `ipfs-embed` 0.21

Reply:
`Keypair` doesnt implement `Clone`, which is required for `cached` to work, so it must be new-typed with `impl Clone`